ocp_resources/self_node_remediation_templates.py

```diff
@@ -0,0 +1,47 @@
+# API reference: https://github.com/medik8s/self-node-remediation
+from typing import Any, Union
+
+from ocp_resources.resource import NamespacedResource
+
+
+class SelfNodeRemediationTemplate(NamespacedResource):
+    """
+    SelfNodeRemediationTemplate CRD
+    """
+
+    api_group = NamespacedResource.ApiGroup.SELF_NODE_REMEDIATION_MEDIK8S_IO
+    VALID_STRATEGIES = {"Automatic", "ResourceDeletion", "OutOfServiceTaint"}
+
+    def __init__(
+        self,
+        remediation_strategy: Union[str, None] = "Automatic",
+        **kwargs: Any,
+    ) -> None:
+        """
+        Create SelfNodeRemediationTemplate object.
+
+        Args:
+            remediation_strategy (str, optional): The remediation method for unhealthy nodes.
+                Options are "Automatic", "OutOfServiceTaint", or "ResourceDeletion".
+                Default is "Automatic".
+        """
+        if remediation_strategy not in SelfNodeRemediationTemplate.VALID_STRATEGIES:
+            raise ValueError(
+                f"Invalid remediation strategy: {remediation_strategy}. "
+                f"Supported values are: {', '.join(SelfNodeRemediationTemplate.VALID_STRATEGIES)}"
+            )
```

Comment:
should be done by the resource api; please remove

Reply:
ack, Thanks